vespajlib/src/main/java/com/yahoo/collections/MethodCache.java

```diff
@@ -33,18 +33,23 @@ public void clear() {
     public Method get(Object object) {
         return get(object, null);
     }
+
     public Method get(Object object, Consumer<String> onPut) {
-        Method m = cache.get(object.getClass().getName());
-        if (m == null) {
-            m = lookupMethod(object);
-            if (m != null) {
-                if (onPut != null)
-                    onPut.accept(object.getClass().getName());
-                cache.put(object.getClass().getName(), m);
-            }
+        Pair<Class<?>, Method> pair = cache.get(object.getClass().getName());
+        if (pair != null && pair.getFirst() != object.getClass()) {
```

Comment:
You need a comment similar to the one above to explain the dropping of the cache.

Reply:
```suggestion
        // When changing bundles, you might end up having cached the old method pointing to the old bundle.
        // That will then lead to a class cast exception when invoking the wrong clone method.
        // Whenever we detect a new class with the same name, we therefore drop the entire cache. 
        // This is also the reason for caching the pair of method and original class—not just the method.
        if (pair != null && pair.getFirst() != object.getClass()) {
```